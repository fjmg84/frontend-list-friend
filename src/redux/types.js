export const TYPES = {
  LOAD: "load/friends",
  ERROR_ACTIVE: "error/active",
};
