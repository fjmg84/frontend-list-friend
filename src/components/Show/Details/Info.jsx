import React from "react";
import PropTypes from "prop-types";

export const Info = ({ friend = {} }) => {
  return (
    <>
      <div className="article col">
        <p className="details-title">Bio:</p>
        <div className="details-text">{friend.bio}</div>
      </div>
      <div className="divider"></div>

      <div className="article row">
        <p className="details-title">Phone:</p>
        <div className="details-text">{friend.phone}</div>
      </div>
      <div className="divider"></div>

      <div>
        <div className="article row">
          <p className="details-title">Address</p>
          <div className="details-text">{friend.address_1}</div>
        </div>

        <div className="article row">
          <p className="details-title">City</p>
          <div className="details-text">{friend.city}</div>
        </div>

        <div className="article row">
          <p className="details-title">Zipcode</p>
          <div className="details-text">{friend.zipcode}</div>
        </div>

        <div className="article row">
          <p className="details-title">State</p>
          <div className="details-text">{friend.state}</div>
        </div>
      </div>
    </>
  );
};

Info.protoTypes = {
  friend: PropTypes.object,
};

Info.propTypes = {
  friend: PropTypes.object.isRequired,
};
