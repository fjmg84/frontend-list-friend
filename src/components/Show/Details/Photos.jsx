import React, { useState } from "react";
import { MdClear } from "react-icons/md";
import PropTypes from "prop-types";

export const Photos = ({ photos = [] }) => {
  const [modal, setModal] = useState({
    view: false,
    url: undefined,
  });

  return (
    <>
      <div className="article">
        <div className="gallery">
          {photos.map((photo, index) => {
            return (
              <button
                key={index}
                onClick={() =>
                  setModal({
                    view: true,
                    url: photo,
                  })
                }
              >
                <img src={photo} alt={photo} />
              </button>
            );
          })}
        </div>
      </div>

      {modal.view && (
        <div id="miModal" className="modal">
          <div className="icons icons-white close">
            <span
              className="fill vector"
              onClick={() => setModal({ view: false })}
            >
              <MdClear size={24} />
            </span>
          </div>

          <div className="modal-content">
            <img src={modal.url} alt={modal.url} />
          </div>
        </div>
      )}
    </>
  );
};

Photos.protoTypes = {
  photos: PropTypes.array,
};
