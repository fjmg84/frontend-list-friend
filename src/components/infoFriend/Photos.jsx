import React from "react";

export const Photos = ({ photos }) => {
  console.log(photos);
  return (
    <>
      <div className="article">
        <div className="gallery">
          {photos.map((photo, index) => {
            return <img key={index} src={photo} alt={photo} />;
          })}
        </div>
      </div>
    </>
  );
};
