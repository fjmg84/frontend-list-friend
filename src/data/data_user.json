{
    "id": 6,
    "img": "https://s3.amazonaws.com/uifaces/faces/twitter/walterstephanie/128.jpg",
    "first_name": "Steph",
    "last_name": "Walters",
    "phone": "(820) 289-1818",
    "address_1": "5190 Center Court Drive",
    "city": "Spring",
    "state": "TX",
    "zipcode": "77370",
    "bio": "I'm very choosy. I'm also very suspicious, very irrational and I have a very short temper. I'm also extremely jealous and slow to forgive. Just so you know.",
    "photos": [
        "/static/media/photo_1.65b44377ca3ba76ac460.jpg",
        "/static/media/photo_2.f94104513f3ac56066cc.jpg",
        "/static/media/photo_3.a617d38fcaf25508172f.jpg"
    ],
    "statuses": [
        "Developing something amazing",
        "This could be interesting....",
        "Man, life is so good",
        "There is nothing quite like a good friend",
        "Take a look around you, everything is awesome",
        "What is the point of all of this"
    ],
    "available": true
}