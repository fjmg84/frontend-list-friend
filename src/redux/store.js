import { createStore, combineReducers } from "redux";
import { friendsReducer } from "../reducer/friends";
import { errorsReducer } from "../reducer/errors";

const rootReducers = combineReducers({
  friendsReducer,
  errorsReducer,
});

export const store = createStore(
  rootReducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
