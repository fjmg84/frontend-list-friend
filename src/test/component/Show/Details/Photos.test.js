import { Provider } from "react-redux";
import { render } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { store } from "../../../../redux/store";
import { TYPES } from "../../../../redux/types";
import data_user from "../../../../data/data_user.json";
import data from "../../../../data/data.json";
import { Photos } from "../../../../components/Show/Details/Photos";

describe("testing to component <Photos/>", () => {
  test("snapshot to component", () => {
    store.dispatch({
      type: TYPES.LOAD,
      payload: data,
    });

    const wrapper = render(
      <MemoryRouter>
        <Provider store={store}>
          <Photos photos={data_user.photos} />
        </Provider>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
