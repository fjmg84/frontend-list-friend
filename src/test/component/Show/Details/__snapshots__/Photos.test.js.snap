// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing to component <Photos/> snapshot to component 1`] = `
<div
  class="article"
>
  <div
    class="gallery"
  >
    <button>
      <img
        alt="/static/media/photo_1.65b44377ca3ba76ac460.jpg"
        src="/static/media/photo_1.65b44377ca3ba76ac460.jpg"
      />
    </button>
    <button>
      <img
        alt="/static/media/photo_2.f94104513f3ac56066cc.jpg"
        src="/static/media/photo_2.f94104513f3ac56066cc.jpg"
      />
    </button>
    <button>
      <img
        alt="/static/media/photo_3.a617d38fcaf25508172f.jpg"
        src="/static/media/photo_3.a617d38fcaf25508172f.jpg"
      />
    </button>
  </div>
</div>
`;
